contracts/test/4337/Test4337ModuleAndHandler.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+pragma abicoder v2;
+
+import "../../libraries/SafeStorage.sol";
+import "./UserOperation.sol";
+
+interface ISafe {
+    function execTransactionFromModule(address to, uint256 value, bytes memory data, uint8 operation) external returns (bool success);
+}
+
+/// @dev A Dummy 4337 Module/Handler for testing purposes
+/// @dev DO NOT USE IN PRODUCTION
+contract Test4337ModuleAndHandler is SafeStorage {
+    using UserOperationLib for UserOperation;
+
+    address public immutable myAddress;
+    address public immutable entryPoint;
+
+    address internal constant SENTINEL_MODULES = address(0x1);
+
+    constructor(address entryPointAddress) {
+        entryPoint = entryPointAddress;
+        myAddress = address(this);
+    }
+
+    function validateUserOp(UserOperation calldata userOp, bytes32, uint256 missingAccountFunds) external returns (uint256 validationData) {
+        address payable safeAddress = payable(userOp.sender);
+        ISafe senderSafe = ISafe(safeAddress);
+
+        if (missingAccountFunds != 0) {
+            ISafe(senderSafe).execTransactionFromModule(entryPoint, missingAccountFunds, "", 0);
```

Comment:
Why is it necessary to cast the `senderSafe` to `ISafe` again?

Reply:
das Hirngespinst